### PR TITLE
Warn find and fix duplicate emails

### PR DESCRIPTION
**Pull request by phlax**
This PR adds:

 - a utility for finding, fixing and validating duplicate emails
 - a system check for warning about duplicate emails
 - commands to:
    - find_duplicate_emails
    - update_user_email

This is a temporary solution to help users before enforcing email uniqueness in db schema (touch #4035)

### CODE REVIEW

**Comment by dwaynebailey** [comment]
These commands need some doc love

**Comment by phlax** [comment]
> These commands need some doc love

Yep - adding now

File: pootle/checks.py [hunk 1]
```diff
@@ -45,6 +46,27 @@ def _version_to_string(version, significance=None):
...
+            errors.append(checks.Warning(_("There are user accounts with "
+                                           "duplicate emails. This will not "
+                                           "be allowed in Pootle 2.8."),
+                                         hint=_("Try using 'pootle "
+                                                "find_duplicate_emails', and "
+                                                "then update user emails with "
+                                                "'pootle update_user_email "
+                                                "username email'"),
+                                         id="pootle.W017"))
+    except (OperationalError, ProgrammingError):
```
Comment (unho) [inline pootle/checks.py]: What about doing this on `get_duplicate_emails()`.

File: pootle/checks.py [hunk 2]
```diff
@@ -45,6 +46,27 @@ def _version_to_string(version, significance=None):
...
+        if len(get_duplicate_emails()):
+            errors.append(checks.Warning(_("There are user accounts with "
+                                           "duplicate emails. This will not "
+                                           "be allowed in Pootle 2.8."),
+                                         hint=_("Try using 'pootle "
+                                                "find_duplicate_emails', and "
+                                                "then update user emails with "
+                                                "'pootle update_user_email "
+                                                "username email'"),
+                                         id="pootle.W017"))
```
Comment (unho) [inline pootle/checks.py]: Please don't indent too much the args for `checks.Warning`. See a few lines below to see what I mean.

File: pootle/apps/accounts/utils.py [hunk 3]
```diff
@@ -349,3 +350,12 @@ def revert_units_state_changed(self):
                 unit.save()
                 logger.debug("Unit state reverted: %s"
                              % (repr(unit)))
+
+
+def get_duplicate_emails():
+    """Get a list of emails that occur more than once in user accounts
+    """
+    User = get_user_model()
```
Comment (unho) [inline pootle/apps/accounts/utils.py]: Please factor this out to the module.

File: pootle/apps/accounts/management/commands/update_user_email.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class Command(UserCommand):
+    args = "user email"
+    help = "Update user email address"
+
+    def handle(self, *args, **kwargs):
+        super(Command, self).handle(*args, **kwargs)
+        user = self.get_user(args[0])
+        new_email = args[1]
```
Comment (unho) [inline pootle/apps/accounts/management/commands/update_user_email.py]: I think it is worth ensuring this an actual valid email.

Comment (unho) [inline pootle/apps/accounts/management/commands/update_user_email.py]: Just noticed it is done below. I was expecting it to be validated before used, so IMHO my confusion is understandable.

File: pootle/apps/accounts/management/commands/find_duplicate_emails.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        if not duplicates:
+            print "There are no accounts with duplicate emails"
+            return
+
+        for email in duplicates:
+            users = User.objects.filter(email=email)
+            if not email:
+                print "The following users have no email set:"
+            else:
+                print "The following users have the email: %s" % email
```
Comment (unho) [inline pootle/apps/accounts/management/commands/find_duplicate_emails.py]: First, why you are using `print` as a keyword? Second, why not use `self.stdout.write()`?

File: pootle/apps/accounts/utils.py (hunk 3 above)
Comment (phlax) [inline pootle/apps/accounts/utils.py]: I did but it caused start up errors

File: pootle/apps/accounts/management/commands/find_duplicate_emails.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    def handle(self, *args, **kwargs):
+        duplicates = get_duplicate_emails()
+        if not duplicates:
+            print "There are no accounts with duplicate emails"
+            return
+
+        for email in duplicates:
+            users = User.objects.filter(email=email)
+            if not email:
```
Comment (unho) [inline pootle/apps/accounts/management/commands/find_duplicate_emails.py]: How is this scenario even possible? I mean, if you have a list of duplicate emails `email` here should have some value, isn't it?

File: pootle/checks.py (hunk 1 above)
Comment (phlax) [inline pootle/checks.py]: im not sure about this - but i concluded that this error should only ever happen in tests when the checks are run - so i figured that was the most appropriate place for it.

File: pootle/apps/accounts/management/commands/find_duplicate_emails.py (hunk 6 above)
Comment (phlax) [inline pootle/apps/accounts/management/commands/find_duplicate_emails.py]: no - some users have no email set

File: pootle/apps/accounts/management/commands/find_duplicate_emails.py (hunk 5 above)
Comment (phlax) [inline pootle/apps/accounts/management/commands/find_duplicate_emails.py]: this was a first fix - im just in the process of updating

File: pootle/apps/accounts/management/commands/find_duplicate_emails.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **kwargs):
+        duplicates = get_duplicate_emails()
+        if not duplicates:
+            sys.stdout.write("There are no accounts with duplicate emails\n")
```
Comment (unho) [inline pootle/apps/accounts/management/commands/find_duplicate_emails.py]: Why not `self.stdout.write()`?

Comment (phlax) [inline pootle/apps/accounts/management/commands/find_duplicate_emails.py]: i hadnt realised i could - ill update

**Comment by unho** [comment]
lgtm

**Pull request merged**